### PR TITLE
extractor: subscribe to multiple parties

### LINKED ISSUE #1353: Extractor should support multiple participants/actors per database
SPD-1222:

Extractor currently requires you to interact with the ledger API as a single party; ledger API supports receiving the union of events for a set of parties, so let that non-empty set be specified as a parameter to extractor in place of single party.

Touches similar areas to #1352 .

### PR DESCRIPTION
**Pull request by S11001001**
The extractor `--party` option may now specify multiple parties, separated by commas; e.g. instead of `--party Bob` you can say `--party Bob,Bar,Baz` and get the contracts for all three parties in the database.

Fixes #1353.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by leo-da** [comment]
@S11001001 I will give a more thorough review on Tuesday... but so far I don't see changes in the `com.digitalasset.extractor.config.ConfigParser` to support multiple parties. It still says it is a `String`, needs to be `Seq[String]`

```
      opt[String]("party")
        .required()
        .action((x, c) => c.copy(party = x))
        .text("The party whose contract data should be extracted.")
```

**Comment by bitonic** [comment]
@leo-da I think we should keep `--party`. @S11001001 my preference would be to be able to mention `--party` multiple times, because you can have spaces in parties.

**Comment by leo-da** [comment]
@bitonic I don't think it will work with multiple `--party` parameters, it is not defined to support multiple `partiy` params... my understanding it has to be: `opt[Seq[String]]`

**Comment by leo-da** [comment]
> @S11001001 @bitonic I don't think it will work with multiple `--party` parameters, `--party` has to be: `opt[Seq[String]]` to support one or more.

actually it has to be `unbounded`:
```
  /** Allows the argument to appear multiple times. */
  def unbounded(): OptionDef[A, C] = maxOccurs(UNBOUNDED)
```

**Comment by S11001001** [comment]
> my preference would be to be able to mention `--party` multiple times, because you can have spaces in parties.

@francesco-da Not sure how these are related; parties are comma-separated in this PR, not space-separated.

File: extractor/src/main/scala/com/digitalasset/extractor/config/ConfigParser.scala [hunk 1]
```diff
@@ -279,7 +279,7 @@ object ConfigParser {
         cliParams.ledgerPort,
         from,
         to,
-        cliParams.party,
+        ExtractorConfig.parties(cliParams.party),
```
Comment (S11001001) [inline extractor/src/main/scala/com/digitalasset/extractor/config/ConfigParser.scala]: @leo-da This is the splitting you were looking for; I have a weak preference for doing as little parsing into `CliParams` as possible, but do you want me to add _moar type safety_ for `CliParams`?

**Comment by S11001001** [comment]
> Looks good, but please add a tests that ensures we can use parties with spaces in them.

@bitonic Added in b78174d39a8dfbc5fde0899bd8522d3302047557; thanks.

File: extractor/src/main/scala/com/digitalasset/extractor/config/ConfigParser.scala (hunk 1 above)
Comment (leo-da) [reply extractor/src/main/scala/com/digitalasset/extractor/config/ConfigParser.scala]: nah, it is fine... but my main concern was documentation, the help screen. Plus the thing that @bitonic wanted you need `unbounded` option configuration, to support multiple party options.

**Comment by leo-da** [comment]
> > my preference would be to be able to mention `--party` multiple times, because you can have spaces in parties.
> 
> @bitonic Not sure how these are related; parties are comma-separated in this PR, not space-separated.

@S11001001 I think @bitonic wants to support spaces in the Party names like:

`--party "Party A with spaces" --party "Party B with spaces"`
or maybe:
`--party "'Party A with spaces','Party B with spaces'"`

**Comment by S11001001** [comment]
@leo-da I understand that, but as you noted `--party "Party A with spaces,Party B with spaces"` suffices for that, so I'm not sure why spaces-in-parties entails passing the `--party` option multiple times.

File: extractor/src/main/scala/com/digitalasset/extractor/config/ConfigParser.scala (hunk 1 above)
Comment (S11001001) [reply extractor/src/main/scala/com/digitalasset/extractor/config/ConfigParser.scala]: Docced in 1c2cb89d24

**Comment by bitonic** [comment]
@S11001001 I find spaces + commas quite awkward. in any case, this is ok for now.

**Comment by S11001001** [comment]
@leo-da I'd appreciate a re-check as I've changed the design to be closer to what you might have been looking for, so we have better command-line error reporting (doesn't just pass malformed parties through to API). I've also created an exciting new merge conflict for your #1361 🎉

**Pull request merged**